rust/src/platform.rs

```diff
@@ -230,6 +232,101 @@ impl Platform {
     }
 }
 
+pub trait TypeParser {
+    fn parse_types_from_source<S: BnStrCompatible, P: AsRef<Path>>(
+        &self,
+        _source: S,
+        _filename: S,
+        _include_directories: &[P],
+        _auto_type_source: S,
+    ) -> Result<TypeParserResult, String> {
+        Err(String::new())
+    }
+}
+
+#[derive(Clone, Default)]
+pub struct TypeParserResult {
+    pub types: HashMap<String, Ref<Type>>,
+    pub variables: HashMap<String, Ref<Type>>,
+    pub functions: HashMap<String, Ref<Type>>,
+}
+
+impl TypeParser for Platform {
+    fn parse_types_from_source<S: BnStrCompatible, P: AsRef<Path>>(
+        &self,
+        source: S,
+        filename: S,
+        include_directories: &[P],
+        auto_type_source: S,
+    ) -> Result<TypeParserResult, String> {
+        let mut result = BNTypeParserResult {
+            functionCount: 0,
+            typeCount: 0,
+            variableCount: 0,
+            functions: ptr::null_mut(),
+            types: ptr::null_mut(),
+            variables: ptr::null_mut(),
+        };
+
+        let mut type_parser_result = TypeParserResult::default();
+
+        let mut error_string: *mut i8 = ptr::null_mut();
```

Comment:
This actually needs to be a `raw: *mut raw::c_char` per `string::BnString::raw`, and then you can take `BnString::from_raw(...)`; the error strings are left for us to deallocate, plus we need to account for the ARM char difference.

Alternatively, we could just not handle the errors and the fifth argument to `BNParseTypesFromSource` can be a nullptr.  I'm okay with either solution.